card/spec/lib/card/query_spec.rb

```diff
@@ -24,6 +24,11 @@
       is_expected.to eq(%w(C+A D+A F+A))
     end
 
+    it "appends names to card results" do
+      res = Card::Query.run name: [:in, "C", "D", "F", "Z"], append: "A"
+      expect(res).to eq([Card["C+A"],Card["D+A"], Card["F+A"], nil])
+    end
```

Comment:
Our query specs are insufficient. We only test name results.

Reply:
What should a search with an `append` statement return if the card doesn't exist? A new card or nothing? I think returning `nil`, what it currently does, will cause trouble somewhere someday.

Reply:
There should be a `new: {}` there.  I broke it in this commit: https://github.com/xithan/decko/commit/6f1c02a8f7787ab0641385b5d398bd8a49ab86c4#diff-6b22a1f8bd11baaec00638be4a7d3814  (which was merged fairly recently)

Reply:
(only needed in card_result.  `Card.fetch alter_result(record["name"]), new: {}`